### DISCUSSION TITLE
Is there a plan for a "bruno-as-a-library"?

### DISCUSSION THREAD

**Discussion by ouvreboite** [usebruno/bruno#2656] [Ideas]
On postman's side, newman (the CLI) can also be used as a JS SDK/library. This allows to execute postman collections in various Node/JS context (in the browser, has part of a test step, in a webserver, ...)

Basically, having something like this:
```javascript
import { Bruno } from "@usebruno/sdk"; //new 

let request = `
get {
  url: https://example.com
}
 
tests {
  test("should return 200", function() {
    expect(res.status).to.equal(200);
  });
}`;
let output = new bruno().run([request]); //plus options for env, secrets, etc.
```
(this is not a design proposal, just to clarify what I mean by a "library")

**Comment by gopival** [comment]
@helloanoop Hi, Looks like copilot is suggesting the below library, but I'm unable to find this library in NPM registry. Could you please advise me whether this library is available?

```
**const Bruno = require('@usebruno/bruno');**
```

```
npm error code E404
npm error 404 Not Found - GET https://registry.npmjs.org/@usebruno%2fbruno - Not found
npm error 404
npm error 404  '@usebruno/bruno@*' is not in this registry.
```

**Reply by therealshark** [reply]
@gopival I asked openchat and this was its answer:

> It seems like the library you are trying to use, @usebruno/bruno, is not available in the NPM registry. This could be because the package has been removed from the registry or it was never published there in the first place.
>  
> If you believe that this package should exist in the NPM registry, you can try searching for it again after some time to see if it gets re-published. Alternatively, you may need to look for an alternative library that provides similar functionality.
